components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpOperations.java

```diff
@@ -593,7 +593,17 @@ private boolean doStoreFile(String name, String targetName, Exchange exchange) t
                 log.debug("Took {} ({} millis) to store file: {} and FTP client returned: {}",
                         new Object[]{TimeUtils.printDuration(watch.taken()), watch.taken(), targetName, answer});
             }
-
+            
+            // after storing file, we may set chmod on the file
+            String chmod = ((FtpConfiguration) endpoint.getConfiguration()).getChmod();
+            if (ObjectHelper.isNotEmpty(chmod)) {
+                log.debug("Setting chmod: {} on file: {}", chmod, targetName);
+                String command = "chmod " + chmod + " " + targetName;
+                log.trace("Client sendSiteCommand: {}", command);
+                boolean success = client.sendSiteCommand(command);
+                log.trace("Client sendSiteCommand successful: {}", success);
+            }
+            
```

Comment:
I think we need to store the client reply code / string before doing chmod operation, so they are from the operation that performed the upload. So eg move the above code below the code below that set those headers.